DNN Platform/DotNetNuke.ModulePipeline/ModuleControlPipeline.cs

```diff
@@ -0,0 +1,194 @@
+                {
+                    moduleControl.ModuleContext.Configuration = moduleConfiguration;
+                }
+            }
+
+            if (TracelLogger.IsDebugEnabled)
+                TracelLogger.Debug($"ModuleControlFactory.LoadModuleControl End (TabId:{moduleConfiguration.TabID},ModuleId:{moduleConfiguration.ModuleID}): ModuleControlSource:{moduleConfiguration.ModuleControl.ControlSrc}");
+            return control;
+        }
+
+        public Control LoadSettingsControl(TemplateControl containerControl, ModuleInfo moduleConfiguration, string controlSrc)
+        {
+            if (TracelLogger.IsDebugEnabled)
+                TracelLogger.Debug($"ModuleControlFactory.LoadSettingsControl Start (TabId:{moduleConfiguration.TabID},ModuleId:{moduleConfiguration.ModuleID}): ModuleControlSource:{moduleConfiguration.ModuleControl.ControlSrc}");
+
+            Control control = null;
+            IModuleControlFactory controlFactory = GetModuleControlFactory(controlSrc);
+
+            if (controlFactory != null)
+            {
+                control = controlFactory.CreateSettingsControl(containerControl, moduleConfiguration, controlSrc);
+            }
+
+            // set the control ID to the resource file name ( ie. controlname.ascx = controlname )
+            // this is necessary for the Localization in PageBase
+            if (control != null)
+            {
+                var fileNameWithoutExtension = Path.GetFileNameWithoutExtension(controlSrc);
+                if (fileNameWithoutExtension != null)
+                {
+                    control.ID = fileNameWithoutExtension.Replace('.', '-');
+                }
+
+                var settingsControl = control as ISettingsControl;
+
+                if (settingsControl != null)
+                {
+                    settingsControl.ModuleContext.Configuration = moduleConfiguration;
+                }
+            }
+
+            if (TracelLogger.IsDebugEnabled)
+                TracelLogger.Debug($"ModuleControlFactory.LoadSettingsControl End (TabId:{moduleConfiguration.TabID},ModuleId:{moduleConfiguration.ModuleID}): ModuleControlSource:{moduleConfiguration.ModuleControl.ControlSrc}");
+            return control;
+        }
+
+        public Control CreateCachedControl(string cachedContent, ModuleInfo moduleConfiguration)
+        {
+            var moduleControl = new CachedModuleControl(cachedContent);
+            moduleControl.ModuleContext.Configuration = moduleConfiguration;
+            return moduleControl;
+        }
+
+        public Control CreateModuleControl(ModuleInfo moduleConfiguration)
+        {
+            string extension = Path.GetExtension(moduleConfiguration.ModuleControl.ControlSrc.ToLowerInvariant());
+            var moduleControl = new ModuleControlBase();
+            moduleControl.ModuleContext.Configuration = moduleConfiguration;
+
+            switch (extension)
+            {
+                case ".mvc":
```

Comment:
should `IModuleControlFactory` have a `GetLocalResourceFile` method?

Reply:
We can create a `GetLocalResourceFile` to extract a method for this lookup, great idea!

Reply:
I originally thought this was new code and not existing code. I think we should mark this as a follow-up item and create a new GitHub Issue. This is the exact same code that was in the original `ModuleControlFactory` that has been ported to this class.

Reply:
Follow-up item created #2780 marking this as resolved

Reply:
Resolved by opening #2780